lib/modules/manager/gradle/artifacts.ts

```diff
@@ -102,12 +104,25 @@ export async function updateArtifacts({
     );
     return null;
   }
+
+  if (
+    config.isLockFileMaintenance &&
+    (!isGradleBuildFile(packageFileName) ||
+      dirname(packageFileName) !== dirname(gradlewFile))
+  ) {
+    logger.trace(
+      'No build.gradle(.kts) file or not in root project - skipping lock file maintenance'
+    );
+    return null;
+  }
+
   logger.debug('Updating found Gradle dependency lockfiles');
 
   try {
     const oldLockFileContentMap = await getFileContentMap(lockFiles);
 
     await writeLocalFile(packageFileName, newPackageFileContent);
+    await prepareGradleCommand(gradlewFile);
```

Comment:
we should issue a warning when file isn't executable, as it's clearly a user error.